json/src/main/scala/Json.scala

```diff
@@ -1 +1,74 @@
 package com.jpmc.json
+
+sealed abstract trait JSON extends Product with Serializable
+
+object JSON {
+
+  case class String(str: java.lang.String) extends JSON
+
+  case class Boolean(value: java.lang.Boolean) extends JSON
+
+  case class Number(num: Int) extends JSON
+
+  case object Null extends JSON
+
+  def apply(str: java.lang.String): JSON = String(str)
+
+  def apply(bool: java.lang.Boolean): JSON = Boolean(bool)
+
+  def apply(num: Int): JSON = Number(num)
+
+  sealed trait Doc extends JSON
+
+  case class Array(data: List[JSON]) extends Doc
+
+  case class Object(data: Map[java.lang.String, JSON]) extends Doc
+
+  def apply(data: JSON*): Doc = Array(data.toList)
+
+  def prettyPrint(json: JSON): java.lang.String = {
+    val quotes = "\""
+    json match {
+      case jsonDoc: Doc => prettyPrintDoc(jsonDoc)
+      case JSON.String(str) => quotes + str + quotes
+      case JSON.Boolean(bool) => bool.toString
+      case JSON.Number(num) => num.toString
+      case JSON.Null => "null"
+    }
+  }
+
+  private def prettyPrintDoc(jsonDoc: Doc): java.lang.String = {
+    val comma = ", "
+    jsonDoc match {
+      case Array(data) =>
+        val openSquare = "["
+        val closeSquare = "]"
+        openSquare + data.map(prettyPrint).mkString(comma) + closeSquare
+      case Object(data) =>
+        val quotes = "\""
+        val colon = ": "
+        val openCurly = "{"
+        val closeCurly = "}"
+        data.toList.map {
+          case (key, value) => quotes + key + quotes + colon + prettyPrint(value)
+        }.mkString(openCurly, comma, closeCurly)
+    }
+  }
+
+  private def removeNullValues(jsonDoc: Doc): Doc = {
+    jsonDoc match {
+      case Array(data) => Array(data = data.filter(_ != Null))
+      case Object(data) => Object(data = data.collect {
+        case (key, obj: JSON.Object) => key -> removeNullValues(obj)
+        case (key, json: JSON) if json != Null => key -> json
+      })
+    }
+  }
+
+  def prettyPrintNoNulls(json: JSON): java.lang.String = {
```

Comment:
You don't need this function
If you don't want to type a lot in test cases - just swap the parameter type to Doc